src/makeCompleteAction.js

```diff
@@ -69,18 +69,52 @@ const mountRequest = (name, options) => (requestType, payload) => {
     throw new Error("The request type name should be a string.");
 
   const typesOfActions = {
-    request: () => ({
-      type: mountConstant(name, "request", options),
-      payload
-    }),
-    success: () => ({
-      type: mountConstant(name, "success", options),
-      payload
-    }),
-    fail: () => ({
-      type: mountConstant(name, "fail", options),
-      payload
-    })
+    request: () => {
+      if (options && options.customSuccessPayload) {
```

Comment:
You can use the `hasPrefixOrSuffix` function that is in the top of the page and rename it to a better name, like: `checkIfOptionExist`. This function does exactly what you're doing here (verify if an option exists and if so, returns a property).

Reply:
Another thing: please, be aware that you're using `customSuccessPayload` here but this is the request action. You should create a "customRequestPayload", because sometimes we need to send some payload in our request actions too!

We should have the "custom payloads": one for request, another one for success and the last one for failure.

Reply:
done